websocket/websocket.go

```diff
@@ -0,0 +1,115 @@
+package websocket
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	onet "github.com/Jigsaw-Code/outline-ss-server/net"
+	"github.com/gorilla/websocket"
```

Comment:
Let's use https://pkg.go.dev/golang.org/x/net/websocket instead if it provides the features we need.